src/__tests__/core/index.test.js

```diff
@@ -47,19 +49,199 @@ describe('setup', () => {
     const model = mock.calls[0][1].toJS();
     expect(model.auth.redirectUrl).toBe('https://test.com/path/');
   });
+
   it('should work with redirect:false and responseType:id_token', () => {
     const options = {
       auth: {
         redirect: false,
         responseType: 'id_token'
       }
     };
+
     setup('id', 'clientID', 'domain', options, 'hookRunner', 'emitEventFn', 'handleEventFn');
     const { mock } = mockInit;
     expect(mock.calls.length).toBe(1);
     const model = mock.calls[0][1].toJS();
     expect(model).toMatchSnapshot();
   });
+
+  describe('clientBaseUrl', () => {
+    it('should default to the specified domain', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {},
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://my-tenant.us.auth0.com');
+    });
+
+    it('should use the clientBaseUrl option if given', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {
+          clientBaseUrl: 'https://client-base-url.example.com',
+          configurationBaseUrl: 'https://config-base-url.example.com',
+          assetsUrl: 'https://assets-url.example.com'
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://client-base-url.example.com');
+    });
+
+    it('should use configurationBaseUrl if given', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {
+          configurationBaseUrl: 'https://config-base-url.example.com',
+          assetsUrl: 'https://assets-url.example.com'
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://config-base-url.example.com');
+    });
+
+    it('should use assetsUrl if given', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {
+          assetsUrl: 'https://assets-url.example.com'
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://assets-url.example.com');
+    });
+  });
+
+  describe('tenantBaseUrl', () => {
+    it('tenantBaseUrl should default to domain URL when using auth0.com', () => {
```

Comment:
what does "domain URL" mean here? also, you might want to skip repeating yourself "tenantBaseUrl" at the beginning of the text name, as this is inside a describe already scoping that.

Reply:
Thanks, I had meant to change that 👍🏻